APISpec.md

```diff
@@ -0,0 +1,508 @@
+
+### Update a Todo
+---
+
+**Request**:
+```
+PUT /todos/{id}
+```
+
+Headers:
+```
+"Authorization": Bearer <Token>
+```
+
+Body:
+```
+{
+    "meta": {},
+    "data": {
+        "title": "My updated todo",
+        "status": "OPEN|IN_PROGRESS|COMPLETE",
+        "dueDate": "2022-04-02T00:00:00.000Z"
+    }
+}
+```
+
+**Response** - `204 No Content`
+
+**Error Codes**:
+
+`404 Not Found` - Todo with given ID does not exist in the API
+
+`400 Bad Request` - Request payload invalid OR status provided not a valid enum value
+
+`401 Unauthorized` - Request does not have valid authentication to update the todo
+
+### Delete a Todo
+---
+
+**Request**:
+```
+DELETE /todos/{id}
+```
+
+Headers:
+```
+"Authorization": Bearer <Token>
+```
+
+Body: None
+
+**Response** - `204 No Content`
+
+**Error Codes**:
+
+`404 Not Found` - Todo with given ID does not exist in the API
+
+`401 Unauthorized` - Request does not have valid authentication to delete todo
+
+### Create a User
+---
+
+**Request**:
+```
+POST /users
+```
+
+Body:
+```
+{
+    "meta": {},
+    "data": {
+        "username": "tom_jerry",
+        "password": "P@ssw0rd!"
+    }
+}
+```
+
+**Response** - `201 Created`
+
+Headers:
+```
+"Location": "/users/{id}"
+```
+
+**Error Codes**:
+
+`400 Bad Request` - Request payload invalid
+
+`409 Conflict` - User with username already exists in API
+Body:
+```
+{
+    "meta": {
+        "success": false
+    },
+    "data": {
+        "msg": "username already exists"
+    }
+}
+```
+
+### Fetch Users
+---
+
+**Request**:
+```
+GET /users
+```
+
+Body: None
+
+**Response** - `200 OK`
+Body:
+```
+{
+    "meta": {
+        "userCount": 4
+    },
+    "data": [
+        {
+            "id": "3fad06ad-88f5-432d-b91d-0415fdddf015",
+            "username": "one_ok_rock",
+            "password": "hashed password"
+        },
+        {
+            "id": "93383ee7-447f-4896-b60a-8e5a4584810b",
+            "username": "trash",
+            "password": "hashed password"
+        },
+        {
+            "id": "57976ec6-29dc-4125-826b-4c7eddacc6ca",
+            "username": "life_awaits",
+            "password": "hashed password"
+        },
+        {
+            "id": "235d67a7-a511-4bd4-8e85-59ad3c80416d",
+            "username": "beyond",
+            "password": "hashed password"
+        }
+    ]
+}
+```
+
+**Error Codes**: None
+
+### Update a User (only password can be updated)
+---
+
+**Request**:
+```
+PUT /users/{id}/password
+```
+
+Headers: 
+```
+"Authorization": Bearer <Token>
+```
+
+Body:
+```
+{
+    "meta": {},
+    "data": {
+        "oldPassword": "password",
+        "newPassword": "P@ssw0rd!"
+    }
+}
+```
+
+**Response** - `204 No Content`
+
+**Error Codes**:
+
+`404 Not Found` - User with ID does not exist in the API
+
+`401 Unauthorized` - Request does not have valid authentication to update username
+
+`400 Bad Request` - Request payload invalid
+
+`403 Forbidden` - oldPassword does not match password in the API
```

Comment:
I'd say this would be a `400 Bad Request`

Reply:
updated